### PR TITLE
add summariziation

### PR DESCRIPTION
**Pull request by patrickweh**
This pull request adds support for aggregate functions (sum, avg, min, max) for numeric columns in the DataTable class. The $summarizableCols array has been added to allow customization of the columns that can be summarized. By setting $summarizableCols = ['*'], all numeric columns will be eligible for summarization. In the frontend, users can activate these functions through a new Summarize tab in the sidebar.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,64 +28,50 @@ composer require team-nifty-gmbh/tall-datatables
...
+
+Optionally, you can publish and run the migrations with:
 
 ```bash
 php artisan vendor:publish --tag="tall-datatables-migrations"
 php artisan migrate
 ```
 
-You can publish the config file with:
+Optionally, ou can publish the config file with:
```
Comment (SirSplasch) [inline README.md:74]:
```suggestion
Optionally, you can publish the config file with:
```

File: resources/views/livewire/data-table.blade.php [hunk 2]
```diff
@@ -383,6 +391,39 @@ class="w-full"
                             @endif
                         </div>
                     </div>
+                    <div x-cloak x-show="tab === 'summarize'">
+                        <div class="grid grid-cols-1 gap-3">
+                            <template x-for="col in summarizable">
+                                <div>
+
```
Comment (SirSplasch) [inline resources/views/livewire/data-table.blade.php:398]: Leerzeile entfernen

File: src/DataTable.php [hunk 3]
```diff
@@ -57,7 +57,14 @@ class DataTable extends Component
 
     public array $exportColumns = [];
 
-    public bool $isSearchable = false;
+    public array $summarizeCols = [
```
Comment (SirSplasch) [inline src/DataTable.php:60]: Find das wording nicht so gut, fände so was wie calculationCols passender, aber musst wegen mir jetzt nicht anpassen

File: src/DataTable.php [hunk 4]
```diff
@@ -301,16 +344,54 @@ function ($item) use ($returnKeys) {
...
 
         array_pop($this->data['links']);
         array_shift($this->data['links']);
     }
 
+    /**
+     * @param Builder $builder
+     * @return array
+     */
+    public function getSums(Builder $builder): array
```
Comment (SirSplasch) [inline src/DataTable.php:365]: public, dass man das ggf. überschreiben kann?

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: resources/views/livewire/data-table.blade.php (hunk 2 above)
Comment (patrickweh) [reply resources/views/livewire/data-table.blade.php:398]: √

Comment (SirSplasch) [reply resources/views/livewire/data-table.blade.php:398]: nope

File: src/DataTable.php (hunk 3 above)
Comment (patrickweh) [reply src/DataTable.php:60]: hab ich umbenannt

File: src/DataTable.php [hunk 5]
```diff
@@ -301,16 +344,54 @@ function ($item) use ($returnKeys) {
                     );
                 }
             );
+
+            $aggregates = $this->getAggregate($baseQuery);
         }
 
         $result->setCollection($mapped ?? $resultCollection);
```
Comment (SirSplasch) [inline src/DataTable.php:351]:
```suggestion
        $result->setCollection($mapped ?: $resultCollection);
```

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**